metadata_validation_conversion/ontology_improver/views.py

```diff
@@ -127,83 +68,85 @@ def authentication(request):
+        'status': data['status'],
+        'timestamp': datetime.now(tz=timezone.utc),
+        'user': data['user']
+    })
+    url = f"http://backend-svc:8000/data/ontologies/{ontology['key']}"
+    res = requests.get(url)
+    if res.status_code != 200 or len(json.loads(res.content)['hits']['hits']) == 0:
+        return HttpResponse(status=404)
+    es = Elasticsearch([settings.NODE], connection_class=RequestsHttpConnection, \
+                       http_auth=(settings.ES_USER, settings.ES_PASSWORD), \
+                        use_ssl=True, verify_certs=False)
+    if data['status'] == 'Verified':
+        update_payload = {
+            'status_activity': status_activity,
+            'upvotes_count': ontology['upvotes_count'] + 1
+        }
+        es.update(index='ontologies', id=ontology['key'], body={"doc": update_payload})
+    else:
+        update_payload = {
+            'status_activity': status_activity,
+            'downvotes_count': ontology['downvotes_count'] + 1
+        }
+        es.update(index='ontologies', id=ontology['key'], body={"doc": update_payload})
+        # create new record with suggested changes
+        new_ontology = {
+            'id': ontology['id'],
+            'term': ontology['term'],
+            'type': ontology['type'],
+            'key': f"{ontology['term']}-{ontology['id']}",
+            'support': '',
+            'projects': data['project'] if data['project'] else [],
+            'species': ontology['species'],
+            'tags': ontology['tags'],
+            'upvotes_count': 0,
+            'downvotes_count': 0,
+            'status_activity': [{
+                'project': data['project'],
+                'status': 'Awaiting Assessment',
+                'timestamp': datetime.now(tz=timezone.utc),
+                'user': data['user']
+            }]
+        }
+        es.index(index='ontologies', id=new_ontology['key'], body=new_ontology)
+    # update summary stats - increment validated_count
+    for project in data['project']:
+        url = f"http://backend-svc:8000/data/summary_ontologies/{project}"
+        res = requests.get(url)
+        project_stats = json.loads(res.content)['hits']['hits'][0]['_source']
+        project_stats['activity']['validated_count'] = project_stats['activity']['validated_count'] + 1
+        es.index(index='summary_ontologies', id=project, body=project_stats)
+    task = update_ontology_summary.s().set(queue='submission')
+    task_chain = chain(task)
+    res = task_chain.apply_async()
+    return HttpResponse(status=200)
 
 @csrf_exempt
 def get_ontology_details(request, id):
     if request.method != 'GET':
         return HttpResponse("This method is not allowed!\n")
     response = {}
     # get ontology type, status etc from FAANG Ontology Database
-    faang_data = model_to_dict(Ontologies.objects.get(pk=id))
-    del faang_data['verified_by_users']
+    url = f"http://backend-svc:8000/data/ontologies/{id}"
```

Comment:
read this from global constants

Reply:
moved to BE_SVC constant in metadata_validation_conversion/metadata_validation_conversion/constants.py